### PR TITLE
#5053 [Manager] Definitions of Property Dialog are not sorted by orderInGroup

### PR DESCRIPTION
**Pull request by saivirtue**
### Checklist
- [x] [Review Contribution Guidelines](https://ohara.readthedocs.io/en/latest/contributing.html)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #5053)
- [ ] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-manager/client/src/components/common/Definitions/ArrayDef.js [hunk 1]
```diff
@@ -39,8 +39,6 @@ const ArrayDef = props => {
     ...rest
   } = omit(props, ['tableKeys']);
 
-  if (!options) return null;
```
Comment (saivirtue) [inline ohara-manager/client/src/components/common/Definitions/ArrayDef.js:42]: This line will make the `Type.ARRAY` definition disappear...not sure why we need this line before 🤔 
@wu87988622 Could you help me review this one?

File: ohara-manager/client/src/components/common/Definitions/BooleanDef.js [hunk 2]
```diff
@@ -19,33 +19,35 @@ import { omit } from 'lodash';
...
+            onChange={onChange}
+            name={name}
+            inputProps={{ type: 'checkbox', ...restInput }}
+            checked={checked}
+            color="primary"
+          />
+        }
+        label={label}
+      />
+      <FormHelperText children={helperText} />
```
Comment (saivirtue) [inline ohara-manager/client/src/components/common/Definitions/BooleanDef.js:49]: We all love the helper text :)

File: ohara-manager/client/src/components/common/Definitions/RenderDefinition.js [hunk 3]
```diff
@@ -194,7 +194,7 @@ const RenderDefinition = props => {
           return <Field {...ensuredFieldProps} component={Duration} />;
 
         case Type.BINDING_PORT:
-          if (isEmpty(freePorts)) {
+          if (isEmpty(freePorts) || reference === ReferenceEnum.NONE) {
```
Comment (saivirtue) [inline ohara-manager/client/src/components/common/Definitions/RenderDefinition.js:197]: Avoid definition of `Type.BINDING_PORT` but has `Reference.NONE` (which will render a TextField without dropdown list)

**Comment by saivirtue** [comment]
The snapshot of PropertyDialog, for reference:

image

File: ohara-manager/client/src/components/Pipeline/PipelinePropertyDialog/PipelinePropertyDialog.js [hunk 4]
```diff
@@ -80,33 +80,29 @@ const PipelinePropertyDialog = props => {
...
-    });
-
-    return Object.keys(groups).map(group => {
-      return groups[group];
-    });
-  };
-
-  const groups = groupBy(classInfo.settingDefinitions);
+  const definitionsByGroup = _(classInfo.settingDefinitions)
+    // sorting each definition by it's orderInGroup property
```
Comment (konekoya) [inline ohara-manager/client/src/components/Pipeline/PipelinePropertyDialog/PipelinePropertyDialog.js:84]: _sort_ and _its_ :)

File: ohara-manager/client/src/components/common/Definitions/BooleanDef.js (hunk 2 above)
Comment (konekoya) [reply ohara-manager/client/src/components/common/Definitions/BooleanDef.js:49]: So this is where the document and error will be displayed, right 🤔

Comment (saivirtue) [reply ohara-manager/client/src/components/common/Definitions/BooleanDef.js:49]: Hmm...error part 😅

**Pull request merged**